aten/src/ATen/core/NestedIntSymNodeImpl.h

```diff
@@ -183,4 +184,28 @@ class TORCH_API NestedIntSymNodeImpl : public SymNodeImpl {
   int64_t coeff_;
 };
 
+class TORCH_API UnionFind {
+    std::unordered_map<int64_t, std::pair<int64_t, size_t>> data;
```

Comment:
nit: use ska::flat_hash_map, it's faster

Reply:
It would be good to document what the pair means. Better yet, just make a small struct with descriptive names for the fields.

Reply:
Document that this structure is not thread safe